### ISSUE TITLE
Active Tasks are stalling

### ISSUE THREAD

**Issue by daftz** [kieraneglin/pinchflat#468]
It seems that after a while (half a day or so), the active tasks are stalling. For instance, I now see active tasks for almost all my sources and some started 5 hours ago. 

When I restart the docker container the active tasks disappear and appear for a short while when a source is checked for new content, which is normal behaviour I think.

I've attached a screenshot.

Anything I can do to toubleshoot this issue?

<img width="1052" alt="Scherm­afbeelding 2024-11-11 om 16 26 56" src="https://github.com/user-attachments/assets/67251777-8ae6-49be-ac29-0da1fba59e40">

**Comment by kieraneglin** [comment]
Hey there! Thanks for the report (:

That's definitely strange! I don't have any troubleshooting steps off the top of my head, but I'll look in and see what I can find.

For the two `FastIndexingWorker` tasks you've shown here, did those ever finish or do they just hang forever?

**Comment by daftz** [comment]
They seem to hang forever.

Below is the screenshot I just made. It's 08:10 now so those tasks haven been hanging there all night

<img width="459" alt="Scherm­afbeelding 2024-11-13 om 08 08 42" src="https://github.com/user-attachments/assets/44e1f275-89da-4ac3-9b35-73657506fb7d">

**Comment by daftz** [comment]
Would like to add that PF is running on a NAS which is quite busy at certain times (downloading/unpacking things). It might be that PF is locking up because a task is taking too long to complete because of high write/read times on the NAS?

**Comment by daftz** [comment]
Here you go, thanks!

[pinchflat-logs-2024-11-14.txt.zip](https://github.com/user-attachments/files/17747874/pinchflat-logs-2024-11-14.txt.zip)

**Comment by kieraneglin** [comment]
I'm taking a deeper look here and I'm seeing a lot of `Database busy` and database timeout errors in your log file. It looks like the jobs are completing but the database call to mark the job as having completed successfully is failing. On top of that, the queries for loading media items are _extremely_ slow. Given that, a few questions:

- Where are you storing your Pinchflat config directory? Is it on the same machine or a NAS? Is it on an SSD or an HDD?
- How many media items do you have? You can get this by opening a console into the Pinchflat container and running:

```
./bin/pinchflat rpc "Pinchflat.Repo.aggregate(\"media_items\", :count) |> IO.inspect"
```

**Comment by daftz** [comment]
> I'm taking a deeper look here and I'm seeing a lot of `Database busy` and database timeout errors in your log file. It looks like the jobs are completing but the database call to mark the job as having completed successfully is failing. On top of that, the queries for loading media items are _extremely_ slow. Given that, a few questions:
> 
>     * Where are you storing your Pinchflat config directory? Is it on the same machine or a NAS? Is it on an SSD or an HDD?

Yes, on the same NAS. It's a 2 HDD bay Synology. Quite a busy machine at times when downloading/unpacking stuff. I see some 'database locked' messages from time to time in apps like radarr/sonarr but they return to their normal state when the busy times are over.

>     * How many media items do you have? You can get this by opening a console into the Pinchflat container and running:
> 
> 
> ```
> ./bin/pinchflat rpc "Pinchflat.Repo.aggregate(\"media_items\", :count) |> IO.inspect"
> ```
Output is 34529

Thanks again!

**Comment by kieraneglin** [comment]
I've made some recent changes that should fix issues like this! If you still see problems, my hunch is that it comes from running IO-heavy operations (like video downloading) and it's stopping the disk from being able to respond to DB queries in time
